### PR TITLE
Fix usingTLS in idol-library

### PR DESCRIPTION
**Pull request by daniel-lau-mf**
Bump idolVersion to 24.4 for pending release
Patch release all charts
Add basic python tests using helm template

### CODE REVIEW

**Comment by daniel-lau-mf** [comment]
I can/will add the other charts to the test (at least in basic form) but wanted an opinion on whether this looks plausible/useful or not before spending more time on it.

**Comment by hasan-mahmood-opentext** [comment]
Think the test setup looks good. Could add some more checks like testing that setting ingress.tls.secretName adds the tls section to the ingress template, but this definitely looks useful. 
Wonder if we could add some more error handling to things like `render_chart`. If a chart fails to render would be good if the tests could give as much info as possible so they can be more useful for debugging.

**Comment by daniel-lau-mf** [comment]
> Think the test setup looks good. Could add some more checks like testing that setting ingress.tls.secretName adds the tls section to the ingress template, but this definitely looks useful.

Didn't want to go back and try and cover everything we've previously added right now.

> Wonder if we could add some more error handling to things like `render_chart`. If a chart fails to render would be good if the tests could give as much info as possible so they can be more useful for debugging.

[38b00a3](https://github.com/opentext-idol/idol-containers-toolkit/pull/75/commits/38b00a3383379ae581b0b1e5f81a68131e8858ab) adds options to enabled `--validate` and `--debug` which gives a little more output depending on failure mode, though ultimately once something fails I expect devs to start running `helm template` manually to figure out what's gone wrong.

File: helm/test/utils/testbase.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                self.assertIn('apiVersion', obj)
+                if kind not in ret:
+                    ret[kind] = dict()
+                self.assertNotIn(name, ret[kind])
+                ret[kind][name] = obj
+            return ret
+    
+    def test_render_default(self):
+        ''' Chart renders successfully with default values '''
+        objs = self.render_chart()
```
Comment (hasan-mahmood-opentext) [inline helm/test/utils/testbase.py:54]: Is this supposed to use `default_values()`?

Comment (daniel-lau-mf) [reply helm/test/utils/testbase.py:54]: No. I added `default_values()` as a helper for any test that might want to get the default values in future, but they shouldn't need to be passed into `render_chart()`. I should fix `render_chart()` to run with out `--values` if no values are passed in.

**Review by hasan-mahmood-opentext — APPROVED** [review]

**Pull request merged**